auth/maintenance_port_role_manager.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "auth/maintenance_port_role_manager.hh"
+
+#include <seastar/core/future.hh>
+#include <stdexcept>
+#include "log.hh"
+#include "utils/class_registrator.hh"
+
+namespace auth {
+
+constexpr std::string_view maintenance_port_role_manager_name = "com.scylladb.auth.MaintenanceModeRoleManager";
+
+static const class_registrator<
+        role_manager,
+        maintenance_port_role_manager,
+        cql3::query_processor&,
+        ::service::migration_manager&> registration(sstring{maintenance_port_role_manager_name});
+
+
+std::string_view maintenance_port_role_manager::qualified_java_name() const noexcept {
+        return maintenance_port_role_manager_name;
+}
+
+const resource_set& maintenance_port_role_manager::protected_resources() const {
+        static const resource_set resources{};
+
+        return resources;
+}
+
+future<> maintenance_port_role_manager::start() {
+        return make_ready_future<>();
+}
+
+future<> maintenance_port_role_manager::stop() {
+        return make_ready_future<>();
+}
+
+future<> maintenance_port_role_manager::create(std::string_view role_name, const role_config&) {
+        return make_exception_future<>(
+        std::runtime_error("CREATE operation is not supported by MaintenanceModeRoleManager"));
```

Comment:
"Mode" word appearing everywhere.

Reply:
Also, perhaps we should make the error messages more high-level / user-friendly? "RoleManager" Such as "role manager: CREATE operation not supported through maintenance port" (/"socket").